main.js

```diff
@@ -1,65 +1,73 @@
-console.log(affirmations.length); //13
-console.log(mantras.length); //15
 
 var currentMessage = "";
+var usedAffirmations = [];
+var usedMantras = [];
 
 var radioSelection = document.querySelectorAll('input[type="radio"]');
 var messageButton = document.querySelector('.messageButton');
 var message = document.querySelector('.message');
 var image = document.querySelector('img');
 
-window.addEventListener('load', preventDefault);
-
-function preventDefault() {
-  event.preventDefault();
-}
-
 messageButton.addEventListener('click', generateMessage);
-//how can I account for having someone just clicking the button?
 
 function generateMessage() {
-  var affirmationsCopy = affirmations;
-  var mantrasCopy = mantras;
-  //trying to work around the DOM & repopulate an empty array from using the splice method
-  var usedAffirmation = getRandomIndex(affirmations);
-  var usedMantra = getRandomIndex(mantras);
-
+  var affirmationsIndex = getRandomIndex(affirmations);
+  var mantrasIndex = getRandomIndex(mantras);
   if (radioSelection[0].checked) {
-    currentMessage = affirmations[usedAffirmation];
-    manageMessagesDisplayed(affirmationsCopy, affirmations);
-    affirmationsCopy.splice(usedAffirmation, 1);
-    console.log('this is the affirmations array', affirmations, 'this is the current message',
-      currentMessage);
+    if (affirmations.length > 1) {
+      currentMessage = affirmations[affirmationsIndex];
+      message.innerText = currentMessage;
+      var singularAffirmation = affirmations.splice(affirmationsIndex, 1);
+      usedAffirmations.push(singularAffirmation[0]);
+      revealMessage();
+    } else {
+      currentMessage = affirmations[0];
+      message.innerText = currentMessage;
+      var singularAffirmation = affirmations.splice(0, 1);
+      usedAffirmations.push(singularAffirmation[0]);
+      affirmations = usedAffirmations;
+      usedAffirmations = [];
+      revealMessage();
+      theEnd();
+      hideMessage();
+      // manageMessagesDisplayed(affirmations, usedAffirmations);
+    }
+  } else if (radioSelection[1].checked) {
+    if (mantras.length > 1) {
+      currentMessage = mantras[mantrasIndex];
+      var singularMantra = mantras.splice(currentMessage, 1);
+      usedMantras.push(singularMantra[0]);
+      message.innerText = currentMessage;
+      revealMessage();
+    } else {
+      currentMessage = mantras[0];
+      var singularMantra = mantras.splice(currentMessage, 1);
+      usedMantras.push(singularMantra[0]);
+      message.innerText = currentMessage;
+      mantras = usedMantras;
+      usedMantras = [];
+      revealMessage();
+      theEnd();
+      hideMessage();
+      // manageMessagesDisplayed(mantras, usedMantras);
+    }
```

Comment:
This is quite a large function with a lot of repeating code. You could collapse the mantra vs affirmations workflow by creating a variable at the start that determines which arrays you will use. So define two variables at the top of this function `var usedMessages` and `var messages`. Then do an if statement to determine what those two variables will equal. If `radioSelection[0].checked` then `usedMessages` equals `usedAffirmations` and `messages` equal `affirmations`. Else your variables will equal the two mantra arrays. Then you can perform the rest of the functionality using the `usedMessages` and `messages` array. This will decrease your whole function by only needing to do lines 20 through 37 once.